Tasks/CmdLine/cmdline.ts

```diff
@@ -10,9 +10,33 @@ async function run() {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
         // Get inputs.
-        let failOnStderr = tl.getBoolInput('failOnStderr', false);
-        let script: string = tl.getInput('script', false) || '';
-        let workingDirectory = tl.getPathInput('workingDirectory', /*required*/ true, /*check*/ true);
+        let input_failOnStderr = tl.getBoolInput('failOnStderr', false);
+        let input_workingDirectory = tl.getPathInput('workingDirectory', /*required*/ true, /*check*/ true);
+        let input_filePath: string;
+        let input_arguments: string;
+        let input_script: string;
+        let input_targetType: string = tl.getInput('targetType') || '';
+        if (input_targetType.toUpperCase() == 'FILEPATH') {
+            input_filePath = tl.getPathInput('filePath', /*required*/ true);
+            if (!tl.stats(input_filePath).isFile()) {
+                throw new Error(tl.loc('JS_InvalidFilePath', input_filePath));
+            }
+
+            input_arguments = tl.getInput('arguments') || '';
+        }
+        else {
+            input_script = tl.getInput('script', false) || '';
+        }
+
+        // Generate the script contents.
+        let contents: string;
+        if (input_targetType.toUpperCase() == 'FILEPATH') {
+            contents = `. '${input_filePath.replace("'", "''")}' ${input_arguments}`.trim();
```

Comment:
i'm not sure whether it makes sense to dot source for the "Command Line" task. For the "Bash Script" task I think it's OK to just always dot source. My original motivation was to avoid files having to be +x. I need to play around with this a bit more and see whether it matters, or maybe we only dot source if it ends with .sh.